### ISSUE TITLE
When using the default arguments, what is the criterion for fairseq-train to stop training?

### ISSUE THREAD

**Issue by Hbenmazi** [pytorch/fairseq#3186]
#### What is your question?
When using the default arguments, that is, when (--max-epoch, --max-update, --stop-time-hours, --stop-min-lr) are not set, when will fairseq-train stop training?
#### What's your environment?

 - fairseq Version (0.9.0):
 - PyTorch Version (1.5)
 - OS (Linux):
 - How you installed fairseq (source):
 - Build command you used (if compiling from source):pip Install -e
 - Python version: 3.7

**Comment by lematt1991** [comment]
Try it out.  In general it is not recommended to not set any of these parameters

**Comment by Hbenmazi** [comment]
> Try it out. In general it is not recommended to not set any of these parameters

But how to estimate the time required for model training?

**Comment by lematt1991** [comment]
It depends on your training data.  If you feel that it has stopped prematurely (i.e. your loss is still improving), then you can start another training run and resume from your latest checkpoint using the `--restore-file <path to checkpoint>` flag.  If you see that your model is still training, but your loss/validation metrics aren't improving, then you can kill the training job.
